### PR TITLE
WT-6675 Remove Java API

### PR DESCRIPTION
**Pull request by kommiharibabu**

### CODE REVIEW

File: src/docs/tools/doxfilter [hunk 1]
```diff
@@ -27,4 +27,4 @@
 
 tooldir=`dirname $0`
 
-python $tooldir/doxfilter.py "$@"
+cat "$@" | python $tooldir/doxfilter.py
```
Comment (sueloverso) [inline src/docs/tools/doxfilter:30]: While I see below that you removed a lot of code from `doxfilter.py`, the change above is the only one that is not completely straightforward. I see below that you removed the loop processing `sys.argv[]` and instead just use `stdin`. That change seems unrelated to java.

**Review by sueloverso — APPROVED** [review]
I have one minor question otherwise LGTM.

File: build_posix/configure.ac.in [hunk 2]
```diff
@@ -140,26 +140,14 @@ if test "$wt_cv_enable_strict" = "yes"; then
 	AM_CFLAGS="$AM_CFLAGS $wt_cv_strict_warnings"
 fi
 
-# Java and Python APIs
-if test "$wt_cv_enable_java" = "yes" -o "$wt_cv_enable_python" = "yes"; then
+# Python APIs
```
Comment (ddanderson) [inline build_posix/configure.ac.in:143]: nit: should say `Python API` (2 places in this file).

File: dist/s_docs [hunk 3]
```diff
@@ -145,14 +145,13 @@ EOF
...
 	EXTRA_INPUT=""
 	for f in $EXTRAS ; do
 		[ -e "$f" ] && EXTRA_INPUT="$EXTRA_INPUT ../$f"
 	done
 
-	# Run again to generate the full doc set with Python and Java.
+	# Run again to generate the full doc set with Python.
 	[ "$additional_languages" -eq 1 ] && [ "x$EXTRA_INPUT" != "x" ] && (
 	cd ../src/docs &&
 		(eval cat Doxyfile $filter ; cat <<EOF
```
Comment (ddanderson) [inline dist/s_docs:157]: Since EXTRAS is null, I think the second run of doxygen becomes a no-op - that is, same as the first run.  (Despite the comment, we don't generate any extra doc for python).  It's probable that from `EXTRAS=` to `doxygen -` can be deleted.  To verify that, it would be good compare the docs directory with that change vs docs directory generated with your current branch.

File: src/docs/cursor-log.dox [hunk 4]
```diff
@@ -25,11 +25,7 @@ To open a log cursor on the database:
 @snippet ex_log.c log cursor open
 
 A log cursor's key is a unique log record identifier, plus a
-@m_if{c}
 uint32_t
```
Comment (ddanderson) [inline src/docs/cursor-log.dox:28]: While you're in here, please change this to `\c uint32_t`

**Review by ddanderson — APPROVED** [review]
LGTM.  Noted a few improvements to consider.  Nice to clean up the doc sources a bit!

File: src/docs/cursor-log.dox (hunk 4 above)
Comment (kommiharibabu) [reply src/docs/cursor-log.dox]: Fixed.

File: dist/s_docs (hunk 3 above)
Comment (kommiharibabu) [reply dist/s_docs]: I didn't find any difference, so removed the second execution.

File: build_posix/configure.ac.in (hunk 2 above)
Comment (kommiharibabu) [reply build_posix/configure.ac.in]: Fixed.

File: src/docs/tools/doxfilter (hunk 1 above)
Comment (kommiharibabu) [reply src/docs/tools/doxfilter]: Reverted the change to use `sys.arg[]` as earlier.
